tchannel/sync/client.py

```diff
@@ -104,27 +109,48 @@ def __init__(
         )
         self._threadloop = threadloop or ThreadLoop()
 
+        self.__init__ = self._wrap(self.__init__)
         self.advertise = self._wrap(self.advertise)
 
-        self.raw = self._wrap(self.raw)
-        self.thrift = self._wrap(self.thrift)
-        self.json = self._wrap(self.json)
+        self.raw = TChannelSyncWrapper(self.raw, self._threadloop)
+        self.thrift = TChannelSyncWrapper(self.thrift, self._threadloop)
+        self.json = TChannelSyncWrapper(self.json, self._threadloop)
+
+    def register(self, *args, **kwargs):
+        _register(*args, **kwargs)
 
     def _wrap(self, f):
         assert callable(f)
 
         def wrapper(*a, **kw):
-            if not self._threadloop.is_ready():
-                self._threadloop.start()
-
-            future = self._threadloop.submit(
-                lambda: f(*a, **kw)
-            )
-            return future
+            return _sync_to_async(self._threadloop, f, *a, **kw)
 
         return wrapper
 
-    def register(self, *a, **kw):
-        raise NotImplementedError(
-            "Registration not yet supported for sync clients",
-        )
+
+class TChannelSyncWrapper(object):
+    """Wrapper for the API that in the async TChannel class."""
+    def __init__(self, scheme, threadloop):
+        self.scheme = scheme
+        self._threadloop = threadloop
+
+    def __call__(self, *args, **kwargs):
+        return _sync_to_async(self._threadloop, self.scheme, *args, **kwargs)
+
+    def register(self, *args, **kwargs):
+        return _register(*args, **kwargs)
+
+
+def _register(*args, **kwargs):
+    log.warning("Registration not yet supported for sync tchannel.")
+
+    def decorator(fn):
+        pass
+
+    return decorator
+
+
+def _sync_to_async(threadloop, func, *args, **kwargs):
```

Comment:
Well, technically it's async to sync. I would just call it `_submit` or something. Although, I'm not sure you even need this since you can just do the check and submit manually in two places you're calling this.